ngx_http_sxg_filter_module.c

```diff
@@ -429,6 +450,27 @@ static bool copy_buffer_to_sxg_buffer(const ngx_chain_t* in, sxg_buffer_t* buf,
   return true;
 }
 
+static bool make_chain_from_buffer(ngx_http_request_t* req,
+                                   const sxg_buffer_t* src,
+                                   ngx_chain_t** dst) {
+  ngx_buf_t* b = ngx_calloc_buf(req->pool);
+  u_char* copied_buffer = ngx_palloc(req->pool, src->size);
+  ngx_chain_t* out = ngx_alloc_chain_link(req->pool);
+  if (b == NULL || copied_buffer == NULL || out == NULL) {
+    ngx_pfree(req->pool, b);
+    ngx_pfree(req->pool, copied_buffer);
```

Comment:
Should this call `ngx_pfree(req->pool, out)`?

Reply:
Done, thank you.